src/parser.cpp

```diff
@@ -159,7 +159,7 @@ namespace ada {
     }
   }
 
-  ada::URL Parser::getURL() {
+  ada::URL Parser::get_url() {
```

Comment:
I think that it would more consistent if the `Parser` class was named `parser`. You could also use a more precise name... `url_parser` (though maybe not).

Reply:
I think the class names should start with capital letter. If not, it would be hard to distinguish a namespace from a class. For example: `ada::parser` vs `ada::scheme`

Reply:
We write `std::string_view`, not `std::String_view`.

Reply:
It is `std::filesystem::current_path` not `std::filesystem::Current_path`.

Reply:
👍 That's true.